tests/FsCheck.Test/Gen.fs

```diff
@@ -119,6 +119,12 @@ module Gen =
         Gen.listOfLength length (Gen.constant v)
         |> sample1
         |> ((=) (List.init length (fun _ -> v)))
+
+    [<Property>]
+    let Shuffle (NonEmptySet (xs:Set<int>)) =
+        Gen.shuffle xs
+        |> sample 10
+        |> List.exists ((<>) (Set.toSeq xs))
```

Comment:
Indeed, it turns out that if I strengthen the equality comparison to compare lists (which _do_ have structural equality) instead of sequences, the property ultimately fails:

```F#
[<Property(MaxTest = 100000)>]
let Shuffle (NonEmptySet (xs:Set<int>)) =
    let l = Set.toList xs
    Gen.shuffle l
    |> sample 10
    |> List.map Seq.toList
    |> List.exists ((<>) l)
```

This fails as expected:

```
Test 'FsCheck.Test.Gen.Shuffle' failed: FsCheck.Xunit.PropertyFailedException : 
Falsifiable, after 2 tests (0 shrinks) (StdGen (2120765865,296145085)):
Original:
NonEmptySet (set [0])
```